src/UserDetailsPage/UserDetailsPage.tsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import { useQuery } from "@apollo/client";
+import { USER_DETAILS_QUARY } from "../services/GraphQLOperations";
+
+function UserDetailsPage() {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const id = location.state;
+
+  const { loading, error, data } = useQuery(USER_DETAILS_QUARY, {
+    onError: (error) => {
+      if (error.graphQLErrors != null) {
```

Comment:
```suggestion
      if (error.graphQLErrors?.length > 0) {
```

Se não me engano ele pode vir array vazio tbm.